### PR TITLE
Added ability to dump Vulkan api calls for 0th frame in debug mode

### PR DESCRIPTION
**Pull request by jaremieromer**
Added debug functionality to dump the vulkan api calls for the 0th frame to a text file in the sample's install directory.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -73,3 +73,8 @@ install(DIRECTORY   ${PROJECT_SOURCE_DIR}/source/external/imgui
         DESTINATION include
         PATTERN     "src" EXCLUDE
 )
+
+# Install Vulkan Validation Layer Settings
+install(FILES   ${PROJECT_SOURCE_DIR}/vk_layer_settings.txt
```
Comment (McCallisterRomer) [inline CMakeLists.txt]: Since its installing to the sample dir, I think this should be in the sample's cmake.

File: CMakeLists.txt [hunk 2]
```diff
@@ -73,3 +73,8 @@ install(DIRECTORY   ${PROJECT_SOURCE_DIR}/source/external/imgui
...
+        DESTINATION sample
+)
```
Comment (McCallisterRomer) [inline CMakeLists.txt]: newline

File: source/engine/graphics/api/vulkan/core/Instance.cpp [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 namespace
 {
-constexpr auto g_validationLayers = std::array<const char*, 1>{ "VK_LAYER_KHRONOS_validation" };
+constexpr auto g_validationLayers = std::array<const char*, 2>{ "VK_LAYER_KHRONOS_validation", "VK_LAYER_LUNARG_api_dump" };
```
Comment (McCallisterRomer) [inline source/engine/graphics/api/vulkan/core/Instance.cpp:11]: Neat, does this just automatically look for the file in the current directory? Is it well behaved if we don't have the file, but still want validation layers?

File: vk_layer_settings.txt [hunk 4]
```diff
@@ -0,0 +1,136 @@
+# The settings in this file can be used to configure the behavior of
```
Comment (McCallisterRomer) [inline vk_layer_settings.txt]: Should this go inside some directory? sample or script?

**Review by McCallisterRomer — APPROVED** [review]

File: vk_layer_settings.txt (hunk 4 above)
Comment (jaremieromer) [reply vk_layer_settings.txt]: Wasn't sure where the best place was. I like sample

File: source/engine/graphics/api/vulkan/core/Instance.cpp (hunk 3 above)
Comment (jaremieromer) [reply source/engine/graphics/api/vulkan/core/Instance.cpp:11]: Yes it does look in the current dir. It more or less is well behaved (it defaults to stdout, so it's a lot of text on the console, though only when you have validation layers enabled which is already signing up for text in stdout)

**Pull request merged**